src/rgw/rgw_rest_s3.cc

```diff
@@ -3667,7 +3676,7 @@ int RGW_Auth_S3::authorize_v4(RGWRados *store, struct req_state *s)
       }
     }
     string token_value = string(t);
-    if (using_qs && (token == "host")) {
+    if (!s->aws4_auth_presigned_url_bugfix && using_qs && (token == "host")) {
```

Comment:
Same comment as above - Instead of using s->aws_auth_presigned_url_bugfix, store->ctx()->_conf->rgw_s3_auth_aws4_presigned_url_bugfix can be used here.